### PR TITLE
Add option to stop target instance

### PR DESCRIPTION
**Pull request by dfjxs**
Added an option to the GCP collection recipes to also stop the target instance after collecting disks

### CODE REVIEW

File: dftimewolf/lib/collectors/gcloud.py [hunk 1]
```diff
@@ -89,6 +92,24 @@ def Process(self) -> None:
...
+              self.remote_instance_name)
+          # TODO(dfjxs): Account for GKE Nodes
+          remote_instance.Stop()
+        except RuntimeError as exception:
+          self.ModuleError(str(exception), critical=False)
+        self.logger.success(
+            'Stopped instance {0:s}'.format(self.remote_instance_name))
+      else:
+        self.logger.warning(
+            'Could not stop instance, instance name not provided')
```
Comment (tomchop) [inline dftimewolf/lib/collectors/gcloud.py:108]: Should we add this warning to the SetUp phase? Maybe even bail if the user provides that flag and no instance name?

File: dftimewolf/lib/collectors/gcloud.py [hunk 2]
```diff
@@ -89,6 +92,24 @@ def Process(self) -> None:
...
       self.state.StoreContainer(container)
 
+    if self.stop_instance and not warned:
+      if self.remote_instance_name:
+        try:
+          remote_instance = self.remote_project.compute.GetInstance(
+              self.remote_instance_name)
+          # TODO(dfjxs): Account for GKE Nodes
+          remote_instance.Stop()
+        except RuntimeError as exception:
```
Comment (tomchop) [inline dftimewolf/lib/collectors/gcloud.py:102]: Can we catch a narrower exception here?

**Review by tomchop — CHANGES_REQUESTED** [review]

File: dftimewolf/lib/collectors/gcloud.py (hunk 1 above)
Comment (dfjxs) [reply dftimewolf/lib/collectors/gcloud.py:108]: Ah, good point. Will do.

File: dftimewolf/lib/collectors/gcloud.py (hunk 2 above)
Comment (dfjxs) [reply dftimewolf/lib/collectors/gcloud.py:102]: Updated.

**Review by tomchop — APPROVED** [review]

**Pull request merged**